### PR TITLE
Tmapv4

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Heads-up @Nowosad I've opened this draft PR as a placeholder to track changes and to send to others.

File: 09-mapping.Rmd [hunk 1]
```diff
@@ -593,43 +593,42 @@ nz_map = tm_shape(nz) + tm_polygons() +
   tm_layout(bg.color = "lightblue")
 ```
 
+One of the main differences between regular charts (e.g., scatterplots) and maps is that the input data determine the aspect ratio of maps.
+Thus, in this case, we need to calculate the aspect ratios of our two main datasets, `nz_region` and `nz`.
+The following function, `norm_dim()` returns the normalized width (`"w"`) and height (`"h"`) of the object (as `"snpc"` units understood my the graphic device).
```
Comment (Robinlovelace) [inline 09-mapping.Rmd:598]: Awesome.

**Comment by Robinlovelace** [comment]
Heads-up @Nowosad I resolved some conflicts. See commit above...

**Review by Robinlovelace — COMMENTED** [review]
Great work @Nowosad, edging towards ready I sense! Let me know if there's any way I can help with this PR.

**Comment by Nowosad** [comment]
Hi @Robinlovelace -- thanks. It is going fine at the moment. I still need to work on a few things (e.g., exercises). 

Other than that:
- We need to wait for some updates from Martijn (see https://github.com/r-tmap/tmap/issues?q=is%3Aissue+is%3Aopen+label%3Atmap_v4)
- If you want you can try to add an example of using **rayshader** at the end of this chapter. I've played with some **rayshader** code in the last few days, but was unable to create a nice, short example.

**Comment by Robinlovelace** [comment]
Thought on merging this Jakub? I think this is ready to go and that the sooner we get all these improvements onto the main branch the better. We can work around compatibility issues.

**Comment by Nowosad** [comment]
@Robinlovelace I think this is ready for merge -- I assume there will be still some issues, but as you said "the sooner we get all these improvements onto the main branch the better". I plan to merge it this afternoon/tomorrow morning.

**Review by Robinlovelace — COMMENTED** [review]
Epic PR, great work Jakub!

**Pull request merged**